internal/processor/filterset/regexp/regexpfilterset.go

```diff
@@ -26,9 +26,10 @@ import (
 // regexpFilterSet encapsulates a set of filters and caches match results.
 // Filters are re2 regex strings.
 type regexpFilterSet struct {
-	regexes      map[string]*regexp.Regexp
-	cacheEnabled bool
-	cache        *lru.Cache
+	regexes           map[string]*regexp.Regexp
+	cacheEnabled      bool
+	cache             *lru.Cache
+	fullMatchRequired bool
```

Comment:
Not clear why this flag is needed and why the user can't achieve the same by using regex anchors.

Reply:
Yes, this is not strictly necessary, it's just there for convenience. Turned off and invisible by default.

To me, my default expectation was that I would need a full match. So to match all metrics containing some keyword I would expect to have to use
`.*keyword.*` instead of just `keyword` and would expect `keyword` to only match "keyword" and not "this_contains_keyword". 

For Knative, I will likely end up anchoring all the metric name regexes, so this is a nice-to-have option.

Reply:
I am not sure this is justified. I'd prefer to keep the number of configuration options to minimum. Given that the exact same functionality can be achieved by regexp anchors I don't think we should use this. Regexp behavior with regards to anchoring and full vs substring match is well documented and well known by people use regexp-es. I think the additional function can do more harm than good but confusing people or by requiring extra codelines.

Sorry, I am not convinced this is a good idea.

Maybe you can show examples of how bad it really is without this option? Is extra `.*` in a few places really worse than the separate config option?

Reply:
I think it's fair to limit the configuration surface to prevent confusion, so I removed the option. 

For Knative, we'll actually probably add "$pattern^" to each metric name filter, but that's not that big of a deal.